velox/type/DecimalUtil.cpp

```diff
@@ -105,4 +133,22 @@ std::string DecimalUtil::toString<ShortDecimal>(
   auto decimalType = type->asShortDecimal();
   return formatDecimal(decimalType.scale(), value.unscaledValue());
 }
+
+template <>
+void DecimalUtil::rescaleWithRoundUp(
+    ShortDecimal& output,
+    const int128_t& unscaledValue,
+    const uint8_t fromScale,
+    const std::pair<uint8_t, uint8_t>& toPrecisionScale) {
+  output.setUnscaledValue(rescale(unscaledValue, fromScale, toPrecisionScale));
+}
+
+template <>
+void DecimalUtil::rescaleWithRoundUp(
+    LongDecimal& output,
+    const int128_t& unscaledValue,
+    const uint8_t fromScale,
+    const std::pair<uint8_t, uint8_t>& toPrecisionScale) {
```

Comment:
`getDecimalPrecisionScale` return type is `std::pair<int, int>`

Reply:
But, precision, scale won't grow beyond uint8_t.

Reply:
Sure, and they cannot be negative. But there is no benefit to using uint8 types if saving space is the goal.